openoa/utils/entr/__init__.py

```diff
@@ -415,6 +418,210 @@ def load_reanalysis(conn:EntrConnection, plant_metadata:dict, reanalysis_product
+                               (fct_pandas["entr_tag_name"].isin(["IAVL.DnWh", "IAVL.ExtPwrDnWh"]))]\
+                            .pivot(index="date_time", columns="entr_tag_name", values="tag_value")\
+                            .resample("1M").sum()\
+                            .reset_index()
+
+    #penOA Meter table
+    openoa_meter = fct_pandas[(fct_pandas["table"]=="plant") &\
+                               (fct_pandas["entr_tag_name"].isin(["MMTR.SupWh"]))]\
+                            .pivot(index="date_time", columns="entr_tag_name", values="tag_value")\
+                            .reset_index()
+
+    # Build the OpenOA Plant Data Object
+    project = PlantData(
+        analysis_type="MonteCarloAEP",  # No validation desired at this point in time
+        metadata=plant_meta,
+        meter=openoa_meter,
+        curtail=openoa_curtail,
+        reanalysis={"era5": openoa_era5, "merra2": openoa_merra2},
+    )
+    # Run the MonteCarloAEP
+    pa = MonteCarloAEP(project, reanalysis_products = ['era5', 'merra2'])
+    pa.run(num_sim=1000)
+    # Serialize and return DDL "plant_name string, aep_GWh float"
+    # return pd.DataFrame([{"plant_name": plant_name,
+    #                      "aep_GWh": pa.results.aep_GWh.mean()}]) # TODO: Return the whole series, and other numbers in this structure, as a complex type
+    # TODO: Also return availability and curtaulment estimates
+    return pd.DataFrame([{
+        "plant_name": plant_name,
+        "results": pickle.dumps(pa.results)
+    }])
+
+
+def aep_spark_map_build_metadata(conn:PySparkEntrConnection, plants:list):
+    """
+    Driver-side function to build metadata dictionaries required for OpenOA constructor
+    """
+    import pyspark.sql.functions as f
+
+    spark = conn._conn # Private varaible _conn defined in PySparkEntrConnection class.
+    metadata_dict = {}
+
+    dim_asset_wind_plant = spark.table("dim_asset_wind_plant").filter(f.col("plant_name").isin(plants)).toPandas()
+
+    for plant in plants:
+
+        plant_meta = dim_asset_wind_plant[dim_asset_wind_plant["plant_name"]==plant]
+
+        metadata_dict[plant] = {
+            "latitude": float(plant_meta.latitude),
+            "longitude": float(plant_meta.longitude),
+            "capacity": float(plant_meta.plant_capacity),
+            "number_of_turbines": int(plant_meta.number_of_turbines),
+            "turbine_capacity": float(plant_meta.turbine_capacity),
+            "_entr_plant_id": int(plant_meta.plant_id)
+        }
+
+        metadata_dict[plant]["reanalysis"] = {}
+
+        metadata_dict[plant]["reanalysis"]["era5"] =  {
+            "frequency": "H", #TODO: Read this from Metadata tables
+            "surface_pressure": "WMETR.EnvPres",
+            "temperature": "WMETR.EnvTmp",
+            "time": "date_time",
+            "windspeed_u": "WMETR.HorWdSpdU",
+            "windspeed_v": "WMETR.HorWdSpdV",
+            "density":  "WMETR.AirDen"
+        }
```

Comment:
I see you have TODOs for reading from metadata tables. Probably not a top priority, but how hard would it be to get this information from the ENTR warehouse?